app/master/build.py

```diff
@@ -228,6 +229,27 @@ def complete_subjob(self, subjob_id, payload=None):
             self.mark_failed('Error occurred while completing subjob {}.'.format(subjob_id))
             raise
 
+    def _get_atom_artifact_sys_path(self, subjob_id, atom_id):
+        return os.path.join(
+            self._build_results_dir(),
+            Subjob.ATOM_DIR_FORMAT.format(
+                subjob_id,
+                atom_id,
+            ),
+        )
+
+    def _parse_payload_for_atom_exit_code(self, subjob_id):
+        subjob = self.subjob(subjob_id)
+        for atom_id in range(len(subjob.atoms)):
+            with open(
+                os.path.join(
+                    self._get_atom_artifact_sys_path(subjob_id, atom_id),
+                    Subjob.EXIT_CODE_FILE,
+                ),
```

Comment:
This is minor, but I was wondering what you think -- would this chunk be easier to read if we created the filename in a variable on a previous line?